Created the cast dataset and created by genre graphs

andywang219 opened the pull request:
I cleaned the credits.csv and created a cast dataframe and created a csv as Phil suggested. This dataframe has the actor/actress's name, the movies they've been in, and the character they've played as, as the columns. I need help optimizing the code though. Not sure how to use the apply() method to speed up the changing row values. For the graphs.py, I added two graphs that are related to genres. I believe you need a plot.ly account in order to display the graphs. Please test them out on your machine.

andywang219 committed:
Created the cast dataset and created by genre graphs

I cleaned the credits.csv and created a cast dataframe and created a csv as Phil suggested. This dataframe has the actor/actress's name, the movies they've been in, and the character they've played as, as the columns. I need help optimizing the code though. Not sure how to use the apply() method to speed up the changing row values. For the graphs.py, I added two graphs that are related to genres. I believe you need a plot.ly account in order to display the graphs. Please test them out on your machine.

lihaojin reviewed (approved):
This looks good.

bdinhle reviewed (approved):
Looks great! I'll test out the graphs after you merge. I'm not too sure on .apply() either so maybe Phil or Mohamed can help you out with that

khelif96 commented on datasets.py:
```diff
@@ -142,3 +142,39 @@ def getRatings():
+    # returns a dataset with the name of the actor/actress, movies they've been in, and characters they've played as
+    credits_df = pd.read_csv("data/credits.csv")
+    credits_dict = {}
+    for row in range(len(credits_df)):
```
yeah this is incredibly inefficient looks like it has O(N^2) run time. you can use `.apply(funcName)` here and it will evaluate  each row against the function it will past the row into your function.
```
def cleaningMethod(row):
   cleaningCodeHere


credits_df.apply(cleaningMethod)
```

lihaojin replied:
Since this is only run once against a static size dataframe to generate a csv, it is fine. A side note, it will not be O(N^2) run time because all of these single loops only depend on the static length of the df.

lihaojin replied:
However, as a future reference, use .apply() but for now it will not hinder speed or performance for our purposes

andywang219 committed:
Updating cast_df

I removed the original data column and added a gender column where 2 is male and 1 is female. I also updated the csv file.
